File: leetcode/92 Reverse Linked List II/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        ListNode hair = new ListNode(-1);
        hair.next = head;
        ListNode previous = hair;
        for (int i=0; i<left-1; i++) {
            previous = previous.next;
        }
        ListNode current = previous.next;
        for (int i=left; i<right; i++) { // ³Æ×÷¡¸´©ÕëÒýÏß¡¹£¡£¡£¡
            ListNode nextNode = current.next;
            current.next = nextNode.next;
            nextNode.next = previous.next;
            previous.next = nextNode;
        }
        return hair.next;
    }
}

File: leetcode/92 Reverse Linked List II/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func reverseBetween(head *ListNode, left int, right int) *ListNode {
    hair := &ListNode{Val: -1}
    hair.Next = head
    previous := hair
    for i:=0; i<left-1; i++ {
        previous = previous.Next
    }
    current := previous.Next
    for i := 0; i < right-left; i++ { // ³Æ×÷¡¸´©ÕëÒýÏß¡¹£¡£¡£¡
        nextNode := current.Next
        current.Next = nextNode.Next
        nextNode.Next = previous.Next
        previous.Next = nextNode
    }
    return hair.Next
}